File: src/dataset/c/space_age_Exercism_ThomasZumsteg/gemini/GeminiFlash_space_age_v4.c
#include "space_age.h"

#include <stdint.h>

static const unsigned long planet_year_lengths[] = {
    [mercury] = 7600544,
    [venus] = 19414149,
    [earth] = 31557600,
    [mars] = 59354033,
    [jupiter] = 374355659,
    [saturn] = 929292363,
    [uranus] = 2651370019,
    [neptune] = 5200418560
};

unsigned long get_year_len(enum planet p) {
    if (p >= mercury && p <= neptune) {
        return planet_year_lengths[p];
    } else {
        return 0;
    }
}

double convert_planet_age(enum planet p, unsigned long int age) {
    return (double)age / (double)get_year_len(p);
}

File: src/dataset/typescript/space-age_Exercism-typescript-thewanionly/gemini/GeminiFlash_space-age_v4.ts
enum OrbitalPeriod {
  mercury = 0.2408467,
  venus = 0.61519726,
  earth = 1,
  mars = 1.8808158,
  jupiter = 11.862615,
  saturn = 29.447498,
  uranus = 84.016846,
  neptune = 164.79132
}

const EARTH_SECONDS_PER_YEAR = 31557600;

type Planet = keyof typeof OrbitalPeriod;

const orbitalPeriodValues: { [key in Planet]: number } = {
  mercury: OrbitalPeriod.mercury,
  venus: OrbitalPeriod.venus,
  earth: OrbitalPeriod.earth,
  mars: OrbitalPeriod.mars,
  jupiter: OrbitalPeriod.jupiter,
  saturn: OrbitalPeriod.saturn,
  uranus: OrbitalPeriod.uranus,
  neptune: OrbitalPeriod.neptune,
};

export function age(planet: Planet, seconds: number): number {
  const planetOrbitalPeriod = orbitalPeriodValues[planet];
  const planetAge = seconds / (planetOrbitalPeriod * EARTH_SECONDS_PER_YEAR);
  return Math.round(planetAge * 100) / 100;
}